apps/gitness/src/components-v2/breadcrumbs/useGetBreadcrumbs.ts

```diff
@@ -0,0 +1,12 @@
+import { UIMatch, useMatches } from 'react-router-dom'
+
+import { CustomHandle } from '../../framework/routing/types'
+
+export const useGetBreadcrumbs = () => {
+  const matches = useMatches()
+  const breadcrumbs = matches.filter(match => (match.handle as CustomHandle)?.breadcrumb) as UIMatch<
```

Comment:
casting using `as` is not ideal. it can prevent catching type incompatibility issues

Reply:
`useMatches` is not a generic function, so the only way to be sure that breadcrumbs match the type when passed to `Breadcrumbs` component is to do the type assertion